packages/mail/lib/getNewsletterSettings.js

```diff
@@ -31,13 +34,16 @@ module.exports = async ({
 
   const status = member.status
   const subscriptions = []
-  supportedInterestIds.forEach(interestId => {
-    // only return already configured interests / subscriptions
-    if (interestId in member.interests) {
+  supportedInterestConfigs.forEach(({ interestId, visibleToRoles }) => {
+    // only return visible interests
+    const subscribed = status === MailchimpInterface.MemberStatus.Subscribed
+      ? (!!member.interests[interestId] || false)
+      : false
+    if (subscribed || !visibleToRoles || !visibleToRoles.length || userIsInRoles(user, visibleToRoles)) {
```

Comment:
`visibleToRoles` can be undefined. `userIsInRoles` will default undefined roles argument to an empty array (`userIsInRoles = (user, roles = []) =>...`).

Would argue to keep this concern to `userIsInRoles` fn.



```suggestion
    if (subscribed || userIsInRoles(user, visibleToRoles)) {
```